astropy/io/fits/convenience.py

```diff
@@ -433,15 +434,24 @@ def writeto(filename, data, header=None, output_verify='exception',
                 checksum=checksum)
 
 
-def table_to_hdu(table):
+def table_to_hdu(table, astropy_native=False):
```

Comment:
Indeed, I do think the default should be `False` here to avoid breaking existing users.
I'm unsubscribing from this PR again, it's 272 comments already, oh boy!
If you do want to break backward compat for FITS / Table things or have other important high-level points, please bring it up on the mailing list for discussion.